src/sage/groups/braid.py

```diff
@@ -1657,6 +1721,90 @@ def is_conjugated(self, other):
         """
         l = conjugatingbraid(self, other)
         return bool(l)
+    
+    def pure_conjugating_braid(self, other):
+        r"""
+        Return a pure conjugating braid, i.e. a conjugating braid whose
+        associated permutation is the identity, if it exists.
+
+        INPUT:
+
+        - ``other`` -- a braid in the same braid group as `self`.
+
+        EXAMPLES::
+
+            sage: B = BraidGroup(4)
+            sage: a = B([1, 2, 3])
+            sage: b = B([3, 2,])
+            sage: c = b ^ 12 * a / b ^ 12
+            sage: d1 = a.conjugating_braid(c)
+            sage: print(len(d1.Tietze()))
+            30
+            sage: print(d1.permutation())
+            [3, 4, 1, 2]
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
+            False
+            sage: d2 = a.pure_conjugating_braid(c)
+            sage: print(len(d2.Tietze()))
+            24
+            sage: print(d2.permutation())
+            [1, 2, 3, 4]
+            sage: d2 * c / d2 == a
+            True
+            sage: print(d2)
+            (s0*s1*s2^2*s1*s0)^4
+            sage: print(a.conjugating_braid(b))
+            None
+            sage: print(a.pure_conjugating_braid(b))
+            None
+            sage: a1 = B([1])
+            sage: a2 = B([2])
+            sage: a1.conjugating_braid(a2)
+            s1*s0
+            sage: a1.permutation()
+            [2, 1, 3, 4]
+            sage: a2.permutation()
+            [1, 3, 2, 4]
+            sage: print(a1.pure_conjugating_braid(a2))
+            None
+            sage: (a1^2).conjugating_braid(a2^2)
+            s1*s0
+            sage: print((a1^2).pure_conjugating_braid(a2^2))
+            None
+        """
+        B = self.parent()
+        n = B.strands()
+        p1 = self.permutation()
+        p2 = other.permutation()
```

Comment:
Actually, since you are converting everything to elements of `SymmetricGroup`, it might be better just to either duplicate the `permutation` method (well, it redirects to `coxeter_group_element()`) or by expanding those methods to take an optional group argument. That way you have less transient objects. Although such changes are not needed for a positive review.